models/src/main/java/com/vimeo/networking2/Entity.kt

```diff
@@ -0,0 +1,17 @@
+package com.vimeo.networking2
+
+/**
+ * Represents data that can be uniquely identified by an [Entity.identifier].
+ *
+ * Implementors of this interface must provide a unique identifier.
+ */
+interface Entity {
+
+    /**
+     * The identifier, as a [String], that uniquely identifies this entity.
+     *
+     * @return a [String], or null if the entity does not have an identifier.
+     */
+    val identifier: String?
```

Comment:
After seeing how this will be implemented in the new DTOs, I think that we should rename this property to `resourceKey` (or rename `resourceKey` to `identifier` if you think that's more semantically correct). The reason for this is that the way the `Entity` interface is implemented in each DTO means that the identifier can be changed independently of the resource key. The below code for instantiating a `Category` is valid code:
```kotlin
val category1 = Category(resourceKey = "1", identifier = "2") // identifier == 2
val category2 = Category(resourceKey = "1') // identifier == 1
```
If you think that `identifier` should be separate from the properties of the DTO, then another alternative is to implement the interface as:
```kotlin
data class Category(val resourceKey: String? = null) : Entity {
  override val identifier: String? = resourceKey
}
```

Reply:
I like the second option of having it separate from the properties of the DTO. . The benefit is that the DTO will represent what is returned by the API. Another benefit is that it doesn’t tie down to the resource key as always being the unique identifier.